components/prize_pool/commons/prize_pool.lua

```diff
@@ -0,0 +1,463 @@
+		local prizeDatum = prizeData[prizeTypes[prizeEnum]]
+		assert(prizeDatum, 'Error: Missmatch between prizeData and prizeTypes. Is a prizeType missing in prizeData?')
+		local fieldName = prizeDatum.header
+		if fieldName then
+			args[fieldName .. '1'] = args[fieldName .. '1'] or args[fieldName]
+			for _, prizeValue, index in Table.iter.pairsByPrefix(args, fieldName) do
+				local data = prizeDatum.headerParse(self, prizeValue, args, index)
+				self:addPrize(prizeEnum, index, data)
+			end
+		end
+	end
+
+	return self.prizes
+end
+
+function PrizePool:_readPlacements(args)
+	local currentPlace = 0
+	for placementIndex = 1, math.huge do
+		if not args[placementIndex] then
+			break
+		end
+
+		local placementInput = Json.parseIfString(args[placementIndex])
+		local placement = Placement(placementInput, currentPlace + 1, self.opponentType)
+
+		currentPlace = placement.placeEnd
+		table.insert(self.placements, placement)
+	end
+
+	return self.placements
+end
+
+function PrizePool:setConfig(option, value)
+	self.options[option] = value
+	return self
+end
+
+function PrizePool:addCustomConfig(name, default, func)
+	config[name] = {
+		default = default,
+		func = func
+	}
+	return self
+end
+
+-- TODO extract parts of this to the future Enum class
+local CUSTOM_TYPES_OFFSET = 100
+local CUSTOM_TYPES_USED = 0
+function PrizePool:addCustomPrizeType(enum, data)
+	prizeTypes[enum] = CUSTOM_TYPES_OFFSET + CUSTOM_TYPES_USED
+	CUSTOM_TYPES_USED = CUSTOM_TYPES_USED + 1
+	prizeData[prizeTypes[enum]] = data
+	return self
+end
+
+function PrizePool:addPrize(enum, index, data)
+	assert(prizeTypes[enum], 'addPrize: Not a valid prizeEnum!')
+	assert(Logic.isNumeric(index), 'addPrize: Index is not numeric!')
+	table.insert(self.prizes, {id = enum .. index, enum = prizeTypes[enum], index = index, data = data})
+	return self
+end
+
+function PrizePool:setWidgetInjector(widgetInjector)
+	assert(widgetInjector:is_a(WidgetInjector), "setWidgetInjector: Not a WidgetIjector")
+	self.widgetInjector = widgetInjector
+	return self
+end
+
+function PrizePool:setLpdbInjector(lpdbInjector)
+	-- TODO: Add type check once interface has been created
+	self.lpdbInjector = lpdbInjector
+	return self
+end
+
+function PrizePool:_hasUsdPrizePool()
+	return (Table.any(self.placements, function (_, placement)
+		return placement.hasUSD
+	end)) or (self.options.autoUSD and Table.any(self.prizes, function (_, prize)
+		return prize.enum == prizeTypes.LOCAL_CURRENCY
+	end))
+end
+
+--
+-- Static Functions
+--
+
+--- Remove all non-numeric characters from an input and changes it to a number.
+-- Most commonly used on money inputs, as they often contain , or .
+function PrizePool._parseInteger(input)
+	return tonumber((string.gsub(input, '[^%d.]', '')))
+end
+
+--- Checks if the input matches the format of a date
+function PrizePool._isValidDateFormat(date)
+	if String.isEmpty(date) then
+		return false
+	end
+	return string.match(date, '%d%d%d%d-%d%d-%d%d') and true or false
```

Comment:
date:match(...)

Reply:
Would need a type check beforehand, but I guess that can be added